### PR TITLE
Add NZXT GRID+ V3 support

### PR DESCRIPTION
**Pull request by severgun**
I'm not C# developer and maybe some parts are not optimal, but it works.
Code based on Kraken X3 module. Protocol found at multiple repos on github.
- [x] RPM Sensors
- [x] Voltage Sensors
- [x] Current Sensors
- [x] Power Sensors (just multiplied values of current and voltage)
- [x] Controls
- [x] Noise Sensor **REQUIRES:** https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/814

Default RPM considered as 40%

Second value in package is noise, but I don't know how to interpret it and there is no reading for it in CAM to take it as a reference.
In my environment it goes from 32 to BA(if I bang sensor hardly). 6A if disconnected. Seems like direct HEX->DEC conversion will give dbA. Anyway, there is no Noise type sensors in LHM.

### CODE REVIEW

**Comment by CZonin** [comment]
Never thought I'd see the day. Looking forward to testing it with FanControl!

**Comment by severgun** [comment]
@CZonin 
I did upload dll in my repo. You can try to backup original and replace one in FanControl.

**Comment by CZonin** [comment]
Tried it out earlier and it worked initially during setup but after restarting the app the Grid sensors we're unavailable. Not sure if it's an issue with FanControl or LibreHardwareMonitor.

**Comment by severgun** [comment]
Within one Windows session I don't have such issue.
After reboot yes. There is startup delay option in FanControl config for that.

**Comment by CZonin** [comment]
Looks like the 3rd party option for sensors got unchecked. Got it all working now. Thanks so much!

File: LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+// All Rights Reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using HidSharp;
+
+namespace LibreHardwareMonitor.Hardware.Controller.Nzxt
+{
+    /**
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs:13]: Please convert this to XML documentation.

File: LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+        private const int FansCount = 6;
+        private readonly Sensor[] _rpmSensors = new Sensor[FansCount];
+        private readonly Sensor[] _voltages = new Sensor[FansCount];
+        private readonly Sensor[] _currents = new Sensor[FansCount];
+        private readonly Sensor[] _powers = new Sensor[FansCount];
+        private readonly Sensor[] _pwmControls = new Sensor[FansCount];
+        private readonly Sensor _noise;
+
+        private readonly Control[] _fanControls = new Control[FansCount];
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs:35]: Format this block

File: LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                    ActivateSensor(_powers[i]);
+                    ActivateSensor(_pwmControls[i]);
+
+                    // NZXT GRID does not report current PWM value. So we need to initialize it with some value to keep GUI and device values in sync.
+                    _fanControls[i].SetDefault();
+                }
+                _noise = new Sensor($"GRID Noise", 0, SensorType.Noise, this, Array.Empty<ParameterDescription>(), settings);
+                ActivateSensor(_noise);
+
+                ThreadPool.UnsafeQueueUserWorkItem(ContinuousRead, _rawData);
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs:93]: Create a new thread for this to avoid potential thread pool exhaustion.

**Review by PhyxionNL — CHANGES_REQUESTED** [review]
Few small changes, rest looks good. Can't test this myself as I don't have such a device :)

File: LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs (hunk 2 above)
Comment (severgun) [reply LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs:35]: Don't understand what should I do. I checked other sources in LHM, looks acceptable.

Comment (PhyxionNL) [reply LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs:35]: Format document in VS (it should sort these fields).

Comment (severgun) [reply LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs]: format did nothing. Sorted manually

**Comment by PhyxionNL** [comment]
Blocked by #814

**Comment by eduwalecki** [comment]
Hi guys, Anyone know if this is working for Grid V2? Thanks

**Comment by chimcen** [comment]
Using latest LibreHardwareMonitor nightly build and latest release of fancontrol - GRID+ V3 didn't work. It found 7 Fans (instead of 6) and no one could controlled (didn't found a speed sensor when initializing).
LibreHardwareMonitor itself found the Grid and connected fans and all is working...

Could anyone help?

**Pull request merged**